executor/src/witgen/jit/block_machine_processor.rs

```diff
@@ -0,0 +1,390 @@
+#![allow(unused)]
+use std::{
+    collections::{BTreeSet, HashSet},
+    fmt::Display,
+};
+
+use bit_vec::BitVec;
+use powdr_ast::analyzed::{AlgebraicReference, Identity};
+use powdr_number::FieldElement;
+
+use crate::witgen::{machines::MachineParts, FixedData};
+
+use super::{
+    affine_symbolic_expression::Effect,
+    cell::Cell,
+    witgen_inference::{FixedEvaluator, Variable, WitgenInference},
+};
+
+#[derive(PartialEq, Eq, PartialOrd, Ord, Hash, Clone, Debug)]
+/// A variable for the block machine solver.
+enum CellOrArg {
+    /// A witness cell in the current block machine.
+    Cell(Cell),
+    /// An argument (input or output) of the block machine.
+    Arg(usize),
+}
+
+impl Display for CellOrArg {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            CellOrArg::Cell(cell) => write!(f, "{cell}"),
+            CellOrArg::Arg(i) => write!(f, "args[{i}]"),
+        }
+    }
+}
+
+impl Variable for CellOrArg {
+    fn from_reference(r: &AlgebraicReference, row_offset: i32) -> Self {
+        Self::Cell(Cell::from_reference(r, row_offset))
+    }
+
+    fn to_poly_id(&self) -> Option<powdr_ast::analyzed::PolyID> {
+        match self {
+            CellOrArg::Cell(cell) => cell.to_poly_id(),
+            CellOrArg::Arg(_) => None,
+        }
+    }
+}
+
+/// A processor for generating JIT code for a block machine.
+struct BlockMachineProcessor<'a, T: FieldElement, FE: FixedEvaluator<T>> {
+    fixed_data: &'a FixedData<'a, T>,
+    fixed_evaluator: FE,
```

Comment:
Shouldn't BlockMachineProcessor implement FixedEvaluator (or at least create it)?